utility_network/display-utility-associations/src/main/java/com/esri/samples/display_utility_associations/DisplayUtilityAssociationsSample.java

```diff
@@ -0,0 +1,245 @@
+      
+      // set initial ArcGISMap extent
+      map.setInitialViewpoint(viewPoint);
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+      mapView.getGraphicsOverlays().add(associationsOverlay);
+      
+      // create a controls box
+      VBox controlsVBox = new VBox();
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(255,255,255,0.5)"), CornerRadii.EMPTY,
+        Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setSpacing(5);
+      controlsVBox.setMaxSize(100, 50);
+      controlsVBox.setAlignment(Pos.BASELINE_LEFT);
+      controlsVBox.getStyleClass().add("panel-region");
+      controlsVBox.getChildren().addAll(attachmentLabel, attachmentImageView, connectivityLabel, connectivityImageView);
+      
+      // add the scene view and controls to the stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+      
+      
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+    
+  }
+  
+  private void addAssociation() {
+    try {
+      // Check if the current viewpoint is outside of the max scale.
+      if (mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE).getTargetScale() >= maxScale)
+        return;
+      
+      // Check if the current viewpoint has an extent.
+      Envelope extent = mapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY).getTargetGeometry().getExtent();
+      if (extent == null) {
+        return;
+      }
+      
+      // Get all of the associations in extent of the viewpoint
+      ListenableFuture<List<UtilityAssociation>> associationsFutureResult = utilityNetwork.getAssociationsAsync(extent);
+      
+      associationsFutureResult.addDoneListener(() -> {
+        
+        try {
+          List<UtilityAssociation> associations = associationsFutureResult.get();
+          
+          associations.forEach(association -> {
+            // Check if the graphics overlay already contains the association.
+            if (associationsOverlay.getGraphics().stream().noneMatch(g -> g.getAttributes().containsKey("GlobalId") && g.getAttributes().get("GlobalId") == association.getGlobalId())) {
+              
+              if (association.getGeometry() != null) {
+                // Create and add a graphic for the association.
+                Map<String, Object> attributes = Map.of("GlobalId", association.getGlobalId(), "AssociationType", association.getAssociationType().toString());
+                Graphic newGraphic = new Graphic(association.getGeometry(), attributes);
+                associationsOverlay.getGraphics().add(newGraphic);
```

Comment:
Perhaps we could add a comment to explain the attributes for the renderer (it can be a bit confusing!)
```suggestion
                // create attributes for the graphic so that the renderes display the appropriate symbol
                Map<String, Object> attributes = Map.of("GlobalId", association.getGlobalId(), "AssociationType", association.getAssociationType().toString());
                // create and add the graphic for the association
                Graphic newGraphic = new Graphic(association.getGeometry(), attributes);
                associationsOverlay.getGraphics().add(newGraphic);
```